swapi/components/millennium-falcon.js

```diff
@@ -7,32 +7,58 @@ export default class MillenniumFalcon extends React.PureComponent {
 
 	static propTypes = {
 		onChooseEndpoint: PropTypes.func.isRequired,
+		onExpandToggle: PropTypes.func.isRequired,
+		onMergeLists: PropTypes.func.isRequired,
 		list: PropTypes.arrayOf(PropTypes.shape({
 			url: PropTypes.string.isRequired,
 		})).isRequired,
 		side: PropTypes.string,
 	};
 
 	render() {
-		const { loading, list, onChooseEndpoint } = this.props;
+		const { loading, list, onChooseEndpoint, onExpandToggle, onMergeLists, endpoint } = this.props;
+
+		const listName = endpoint === 'people' ? endpoint : 'films'
 
 		const iconClass = classNames('fa', {
 			'fa-refresh': loading,
 			'fa-spin': loading,
 			'fa-star': !loading,
 		});
 
-		return (<div>
-			<p><i className={iconClass} /> What do you want to see?</p>
-			<div className="btn-group">
-				<button disabled={loading} onClick={() => onChooseEndpoint('people')} className="btn btn-danger">People</button>
-				<button disabled={loading} onClick={() => onChooseEndpoint('films')} className="btn btn-danger">Films</button>
+		return (
+			<div>
+				<p>
+					<i className={iconClass} />
+					{' '}
+					{loading
+						? <span>Loading {endpoint}, please wait...</span>
+						: <span>What do you want to see?</span>
+					}
+					<br />
+					{!this.props.data.films || !this.props.data.people 
```

Comment:
Jeg kan ikke se at componenten modtage en `data` prop ud fra PropTypes, er det en forglemmelse eller var det for at holde den hemmelige knap lidt mere hemmeligt.